pkg/scheduler/framework/parallelize/parallelism.go

```diff
@@ -53,13 +54,29 @@ func chunkSizeFor(n, parallelism int) int {
 
 // Until is a wrapper around workqueue.ParallelizeUntil to use in scheduling algorithms.
 // A given operation will be a label that is recorded in the goroutine metric.
-func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece workqueue.DoWorkPieceFunc, operation string) {
+func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece func(piece int) error, operation string) error {
 	goroutinesMetric := metrics.Goroutines.WithLabelValues(operation)
+	goroutinesDurationMetric := metrics.GoroutinesDuration.WithLabelValues(operation)
+	errCh := NewErrorChannel()
```

Comment:
The things in `error_channel.go` don't need to be exposed now? If Yes, rather we should make them private to prevent the misuse.